.github/workflows/analysis-of-endpoint-connections.yml

```diff
@@ -24,39 +20,68 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Get list of modified files
-        run: |
-          git diff --name-only origin/${{ github.event.pull_request.base.ref }} HEAD > modified_files.txt
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '${{ env.java }}'
+          distribution: 'temurin'
 
-      # Analyze the client sided REST-API calls
-      - name: Set up Node.js
+      - name: Set up node.js
         uses: actions/setup-node@v4
         with:
           node-version: '${{ env.node }}'
 
-      - name: Install and compile TypeScript
+      - name: Parse client sided REST-API calls
         run: |
-          cd supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/
           npm install
-          tsc -p tsconfig.analysisOfEndpointConnections.json
-
-      - name: Run analysis-of-endpoint-connections-client
-        run: |
+          tsc -p supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/tsconfig.analysisOfEndpointConnections.json
           node supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/AnalysisOfEndpointConnectionsClient.js
 
-      - name: Upload JSON file
+      - name: Parse server sided Endpoints
+        run: ./gradlew :supporting_scripts:analysis-of-endpoint-connections:runEndpointParser
+
+      - name: Upload parsing results
         uses: actions/upload-artifact@v4
         with:
-          name: rest-calls-json
-          path: supporting_scripts/analysis-of-endpoint-connections/restCalls.json
+          name: parsing results
+          path: |
+            supporting_scripts/analysis-of-endpoint-connections/endpoints.json
+            supporting_scripts/analysis-of-endpoint-connections/restCalls.json
+
+  Analysis-of-endpoint-connections:
+    needs: Parse-rest-calls-and-endpoints
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
```

Comment:
Not sure if it is a good idea to use `latest` here. I think taking a fixed version might ensure consistent results of the parser better?

Reply:
I want to be consistent with the other workflows, where we also use Ubuntu-latest.